sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -138,7 +138,7 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
    */
   @throws[AnalysisException]("table does not exist")
   override def listColumns(tableName: String): Dataset[Column] = {
```

Comment:
Thank you again, @hvanhovell 

1. Yep. That is the purpose of this PR, to make the contract consistent with other APIs.
2. The existence checking here is redundant because it call other `listColumns`. The callee will check that.